### PR TITLE
LNURL-auth support for LNDHub

### PR DESCRIPTION
**Pull request by Talej**
# Description

Relates to issue: **ZEUS-1045**

Added support for LNURL-auth for LNDHub backend with signature being `lndhub://user:pass` rather than typical message signing.

Also made a slight adjustment on the LnurlAuth view to center the domain inline with all other components.

This pull request is categorized as a:

- [ ] New feature
- [X] Bug fix
- [ ] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Quality assurance 
- [ ] Other

## Checklist
- [X] I’ve run `yarn run tsc` and made sure my code compiles correctly
- [X] I’ve run `yarn run lint` and made sure my code didn’t contain any problematic patterns
- [X] I’ve run `yarn run prettier` and made sure my code is formatted correctly
- [X] I’ve run `yarn run test` and made sure all of the tests pass

## Testing

If you modified or added a utility file, did you add new unit tests?

- [X] No, I’m a fool
- [ ] Yes
- [ ] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [ ] Android
- [X] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [X] LND (REST)
- [X] LND (Lightning Node Connect)
- [ ] Core Lightning (c-lightning-REST)
- [ ] Core Lightning (Spark)
- [ ] Eclair
- [X] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the Zeus [Transfix page](https://www.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [ ] Contributors will need to run `yarn` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `package-lock.json` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

File: views/Receive.tsx [hunk 1]
```diff
@@ -7,14 +7,23 @@ import {
...
 import { LNURLWithdrawParams } from 'js-lnurl';
 import { ButtonGroup, Header, Icon } from 'react-native-elements';
 import { inject, observer } from 'mobx-react';
 import _map from 'lodash/map';
 
+import NfcManager, {
+    NfcEvents,
+    TagEvent,
+    Ndef
+} from 'react-native-nfc-manager';
```
Comment (kaloudis) [inline views/Receive.tsx:23]: I think the edits on this file were all intended for your other PR

File: backends/LndHub.ts [hunk 2]
```diff
@@ -74,8 +74,13 @@ export default class LndHub extends LND {
             invoice: data.payment_request,
             amount: Number(data.amt && data.amt * 1000)
         });
+    lnurlAuth = () =>
+        Promise.resolve({
+            signature: `lndhub://${stores.settingsStore.username}:${stores.settingsStore.password}`
+        });
```
Comment (kaloudis) [inline backends/LndHub.ts:80]: I thought there was an actual `/auth` endpoint we could use

File: backends/LND.ts [hunk 3]
```diff
@@ -336,10 +336,12 @@ export default class LND {
             msg: Base64Utils.btoa(data.msg),
             signature: data.signature
         });
+    lnurlAuth = (r_hash: string) => this.signMessage(r_hash);
```
Comment (kaloudis) [inline backends/LND.ts:339]: love what you've done here, wrapping up the `signMessage` method for the new call

File: views/Receive.tsx (hunk 1 above)
Comment (Talej) [reply views/Receive.tsx:23]: Looks like I branched off that one accidentally. Clearly wasn’t paying attention.

will fix

File: backends/LndHub.ts (hunk 2 above)
Comment (Talej) [reply backends/LndHub.ts:80]: That endpoint was for actual LNDHub auth (fetching an auth token) - didn’t actually do any message signing

Comment (Talej) [reply backends/LndHub.ts:80]: Corrected!

**Review by kaloudis — APPROVED** [review]

**Pull request merged**